blocks/library/quote/index.js

```diff
@@ -30,21 +31,52 @@ registerBlock( 'core/quote', {
 		}
 	},
 
-	controls: [ 1, 2 ].map( ( variation ) => ( {
-		icon: 'format-quote',
-		title: wp.i18n.sprintf( wp.i18n.__( 'Quote style %d' ), variation ),
-		isActive: ( { style = 1 } ) => style === variation,
-		onClick( attributes, setAttributes ) {
-			setAttributes( { style: variation } );
-		},
-		subscript: variation
-	} ) ),
+	controls: [
+		[
+			{
+				icon: 'editor-alignleft',
+				title: wp.i18n.__( 'Align left' ),
+				isActive: ( { align } ) => ! align || 'left' === align,
+				onClick( attributes, setAttributes ) {
+					setAttributes( { align: undefined } );
+				}
+			},
+			{
+				icon: 'editor-aligncenter',
+				title: wp.i18n.__( 'Align center' ),
+				isActive: ( { align } ) => 'center' === align,
+				onClick( attributes, setAttributes ) {
+					setAttributes( { align: 'center' } );
+				}
+			},
+			{
+				icon: 'editor-alignright',
+				title: wp.i18n.__( 'Align right' ),
+				isActive: ( { align } ) => 'right' === align,
+				onClick( attributes, setAttributes ) {
+					setAttributes( { align: 'right' } );
+				}
+			}
+		],
+		[ 1, 2 ].map( ( variation ) => ( {
+			icon: 'format-quote',
+			title: wp.i18n.sprintf( wp.i18n.__( 'Quote style %d' ), variation ),
+			isActive: ( { style = 1 } ) => style === variation,
+			onClick( attributes, setAttributes ) {
+				setAttributes( { style: variation } );
+			},
+			subscript: variation
+		} ) )
+	],
 
 	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { value, citation, style = 1 } = attributes;
+		const { align, value, citation, style = 1 } = attributes;
 
 		return (
-			<blockquote className={ `blocks-quote blocks-quote-style-${ style }` }>
+			<blockquote
+				className={ `blocks-quote blocks-quote-style-${ style }` }
+				style={ align ? { textAlign: align } : null }
```

Comment:
Can I ask for a clarification? Is this for the current quote block or for the continuous Text block? Yep, some mockups could be good :)